channel.rb

```diff
@@ -42,21 +43,25 @@ def only_one_version_packages
 
   def retrieve_packages
     packages = {}
+    puts "Fetching packages for channel https://#{@domain}/#{@channel_name}..."
     channeldata = HTTParty.get("https://#{@domain}/#{@channel_name}/channeldata.json")["packages"]
-    ARCHES.each do |arch|
-      blob = HTTParty.get("https://#{@domain}/#{@channel_name}/#{arch}/repodata.json")["packages"]
-      blob.each_key do |key|
-        version = blob[key]
-        package_name = version["name"]
-
-        unless packages.key?(package_name)
-          package_data = channeldata[package_name]
-          packages[package_name] = base_package(package_data, package_name)
+    benchmark = Benchmark.measure do
+      ARCHES.each do |arch|
+        blob = HTTParty.get("https://#{@domain}/#{@channel_name}/#{arch}/repodata.json")["packages"]
+        blob.each_key do |key|
+          version = blob[key]
+          package_name = version["name"]
+
+          unless packages.key?(package_name)
+            package_data = channeldata[package_name]
+            packages[package_name] = base_package(package_data, package_name)
+          end
+
+          packages[package_name][:versions] << release_version(key, version)
         end
-
-        packages[package_name][:versions] << release_version(key, version)
       end
     end
+    puts "Finished in #{benchmark.real.round(1)} sec: #{packages.to_json.bytesize / 1_000_000}mb of data."
```

Comment:
Can we use the `logger` class in here instead of using `puts` on these statements? http://sinatrarb.com/intro.html#Logging

Reply:
I'll try replacing the `puts` in this app with logger, but logs for some reason don't show up in Docker Compose or GCP (not sure if it's bc Sinatra defaults to STDERR for logging?). I'll give it a shot tho

Reply:
Turns out passing the logger to Channel is surprisingly tricky since Conda is a singleton. I'm tempted to say we should fix logging for conda-api in a different PR.